### PR TITLE
`Try` Result Extension

### PR DESCRIPTION
**Pull request by NathanMagnus**

### CODE REVIEW

File: Functional.Primitives.Extensions/ResultExtensions.cs [hunk 1]
```diff
@@ -129,5 +129,32 @@ public static class ResultExtensions
 
 		public static async Task<Result<TSuccess, TFailure>> FailureIfNone<TSuccess, TFailure>(this Task<Result<Option<TSuccess>, TFailure>> result, Func<TFailure> failureFactory)
 			=> (await result).FailureIfNone(failureFactory);
+
+		public static Result<TSuccess, TFailure> Try<TInputSuccess, TSuccess, TFailure>(this Result<TInputSuccess, TFailure> result, Func<TInputSuccess, TSuccess> successFactory, Func<Exception, TFailure> failureFactory)
```
Comment (JohannesMoersch) [inline Functional.Primitives.Extensions/ResultExtensions.cs]: What about TrySelect?

**Comment by JohannesMoersch** [comment]
TrySelectAsync varient as well please.

File: Functional.Primitives.Extensions/ResultExtensions.cs (hunk 1 above)
Comment (NathanMagnus) [reply Functional.Primitives.Extensions/ResultExtensions.cs]: Changed to `TrySelect` as this is a combination of both Try and Select.

File: Functional.Primitives.Extensions/ResultExtensions.cs [hunk 2]
```diff
@@ -129,5 +129,32 @@ public static class ResultExtensions
...
+		public static Result<TResult, TFailure> TrySelect<TSuccess, TResult, TFailure>(this Result<TSuccess, TFailure> result, Func<TSuccess, TResult> successFactory, Func<Exception, TFailure> failureFactory)
+		{
+			if (successFactory == null)
+				throw new ArgumentNullException(nameof(successFactory));
+
+			if(failureFactory == null)
+				throw new ArgumentNullException(nameof(failureFactory));
+
+			try
+			{
```
Comment (JohannesMoersch) [inline Functional.Primitives.Extensions/ResultExtensions.cs]: Sorry to knitpick, but should we use a Bind here for consistency with the async endpoint?

Comment (NathanMagnus) [reply Functional.Primitives.Extensions/ResultExtensions.cs]: Absolutely!

**Pull request merged**